docs/cmdline-opts/variable.d

```diff
@@ -0,0 +1,35 @@
+c: Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+SPDX-License-Identifier: curl
+Long: variable
+Arg: <name=text/@file>
+Help: Set variable
+Category: curl
+Example: --variable name=smith $URL
+Added: 8.3.0
+See-also: config
+Multi: append
+---
+Set a variable with "name=content" or "name@file" (where "file" can be stdin
+if set to a single dash (-)). The name is a case sensitive identifier that
+must consist of no other letters than a-z, A-Z, 0-9 or underscore. The
+specified content is then associated with this identifier.
+
+The name must be unique within a command line invoke, setting the same
+variable name again will be ignored.
+
+The contents of a variable can be referenced in a later command line option
+when that option name is prefixed with "--expand-", and the name is used as
+"{{name}}" (without the quotes).
+
+When expanding variables, curl can also expand environment variables if the
+name is prefixed with "env:", like in "{{env:name}}".
```

Comment:
Does `{{env:trim}}` mean the environment variable "trim", or the variable "env", trimmed?  A different choice of syntax would avoid ambiguity - perhaps a pipe character "|" to prefix conversions rather than ":"

Reply:
Yes, thanks. Good point. @emanuele6 also brought up the fact that we right now cannot use a variable named `env`.

I avoided the pipe character so far because it has a special meaning to shells. Maybe that should not stand in the way. Using pipes for this purpose seems to also be more common in other situation featuring similar functionality.

Another alternative is to change the `env:` prefix and use something else to make it refer to an environment variable. Like `{{%name}}` or `{{!name}}` for example. I'm avoiding `$name` since then it *really* becomes hard to figure out if the shell expands the name or curl is!

Reply:
I'll switch to `{{%name}}` for environment variables and we can get a feel for how that is. Then decide if that works or not.

Reply:
`%` might not be a good choice for Windows...

Reply:
It also might get confused with the `%` alias for `percent`.
But for that maybe we can just rename `:percent` to `:url` so it is shorter, and ditch the `:%` alias.